python/lsst/ts/gis/component.py

```diff
@@ -64,20 +66,56 @@ async def update_status(self):
         reply = await self.commander.read()
         if reply is not None:
             raw_status = self.commander.get_raw_string(reply)
-            if self.raw_status != raw_status:
-                await self.csc.evt_rawStatus.set_write(status=raw_status)
-                self.log.debug(f"registers={reply.registers}")
-            self.raw_status = raw_status
-            for index, (current_subsystem, old_subsystem) in enumerate(
-                itertools.zip_longest(
-                    reply.registers, self.system_status, fillvalue=None
+            await self.csc.evt_rawStatus.set_write(status=raw_status)
+            self.log.debug(f"registers={reply.registers}")
+            # Since salobj checks the cached data with set_write, only need
+            # to iterate through the current reply.
+            for index, current_subsystem in enumerate(reply.registers):
+                await self.csc.evt_systemStatus.set_write(
+                    index=index, status=current_subsystem
                 )
-            ):
-                if current_subsystem != old_subsystem:
-                    await self.csc.evt_systemStatus.set_write(
-                        index=index, status=current_subsystem
-                    )
-            self.system_status = reply.registers
+            await self.fill_out_fields(raw_status)
+
+    async def fill_out_fields(self, raw_status):
+        """Fill out subsystem event data.
+
+        TBD
+
+        Parameters
+        ----------
+        raw_status: `str`
+            The string array of 1and0's that comprise the status of the GIS's
+            subsystems.
+        """
+        statuses = raw_status.split(" ")
```

Comment:
I think it would be much simpler if you return an array of 1's and 0's in `commander.py` and pass that on to this method. It saves creating a string from an array and then converting it into an array again.